File: Arrays/JumpGame.java
// LeetCode 55 - Jump Game
// Time Complexity: O(n) | Space Complexity: O(1)
public class JumpGame {
    public boolean canJump(int[] nums) {
        int farthest = 0;

        for (int index = 0; index < nums.length; index++) {
            if (farthest < index) {
                return false;
            }
            farthest = Math.max(farthest, index + nums[index]);
        }

        return true;
    }
}

File: Arrays/JumpGame.go
// LeetCode 55 - Jump Game
// Time Complexity: O(n) | Space Complexity: O(1)
package main

func canJump(nums []int) bool {
	farthest := 0

	for index, value := range nums {
		if farthest < index {
			return false
		}
		if index+value > farthest {
			farthest = index + value
		}
	}

	return true
}
